domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java

```diff
@@ -2971,15 +3042,41 @@ public void getConnection(Handler<AsyncResult<PgConnection>> replyHandler) {
   }
 
   void getSQLConnection(Handler<AsyncResult<SQLConnection>> handler) {
+    getSQLConnection(0, handler);
+  }
+
+  void getSQLConnection(int queryTimeout, Handler<AsyncResult<SQLConnection>> handler) {
     getConnection(res -> {
       if (res.failed()) {
         handler.handle(Future.failedFuture(res.cause()));
         return;
       }
-      handler.handle(Future.succeededFuture(new SQLConnection(res.result(), null)));
+      SQLConnection sqlConnection = new SQLConnection(res.result(), null, queryTimeout);
+      if (queryTimeout > 0) {
+        long timerId = vertx.setTimer(queryTimeout, id -> sqlConnection.conn.cancelRequest(ar -> {
+          if (ar.succeeded()) {
+            log.warn(
+                String.format("Cancelling request due to timeout after : %d ms",
+                    queryTimeout));
+          } else {
+            log.warn("Failed to send cancelling request", ar.cause());
+          }
+        }));
+        activeConnectionsTimers.put(sqlConnection, timerId);
```

Comment:
Can you check whether it is possible to save the timerId in sqlConnection so that we can remove activeConnectionsTimer?

Reply:
I don't think that we should include this completely unrelated to SQLConnection class in this  class.

Reply:
We previously stored timeoutTime, why can't we store timeout timerId in SQLConnection?
We already pass the SQLConnection to cancelConnectionTimeoutTimer:
`private void cancelConnectionTimeoutTimer(SQLConnection sqlConnection)`

Reply:
Timeout time is physical (e.g. domain model) attribute, but timerId is synthetic id, which is generated by vert.x

Reply:
@julianladisch, is my explanation ok?

Reply:
timerId will be more reasonable, ok I will add it.

Reply:
done